src/mem/alloc.h

```diff
@@ -401,8 +557,8 @@ namespace snmalloc
 
       handle_message_queue();
 
-      // Free memory of a statically known size. Must be called with an
-      // external pointer.
+      p = pagemap().getp(p);
```

Comment:
This is on the fast path? Ensure it is erased on non-Cheri case.  Perhaps add a comment?